include/quic/address.hpp

```diff
@@ -303,13 +314,19 @@ namespace oxen::quic
     // to ngtcp2_path*
     struct Path
     {
+        friend class TestHelper;
+        friend class Connection;
+
       public:
         Address local;
         Address remote;
 
       private:
         ngtcp2_path _path{local, remote, nullptr};
 
+        void set_new_remote(const ngtcp2_addr& new_remote, uint16_t p);
+        void set_new_local(const ngtcp2_addr& new_local, uint16_t p);
```

Comment:
It feels odd for these to take a port, since there is already a port inside the ngtcp2_addr.  Is there a reason we can't use the one inside `new_remote`/`new_local`?